docker/go-control-plane/testcontrolplane/callbacks.go

```diff
@@ -0,0 +1,145 @@
+package testcontrolplane
+
+import (
+	"context"
+	"log"
+	"os"
+	"sync"
+
+	core "github.com/envoyproxy/go-control-plane/envoy/config/core/v3"
+	discovery "github.com/envoyproxy/go-control-plane/envoy/service/discovery/v3"
+	"github.com/envoyproxy/go-control-plane/pkg/server/v3"
+)
+
+type Callbacks struct {
+	Signal         chan struct{}
+	Debug          bool
+	Fetches        int
+	Requests       int
+	Responses      int
+	DeltaRequests  int
+	DeltaResponses int
+	Filters        map[string]map[string]bool
+	mu             sync.Mutex
+}
+
+var _ server.Callbacks = &Callbacks{}
+
+func (cb *Callbacks) Report() {
+	cb.mu.Lock()
+	defer cb.mu.Unlock()
+	log.Printf("server callbacks fetches=%d requests=%d responses=%d\n", cb.Fetches, cb.Requests, cb.Responses)
+}
+
+func (cb *Callbacks) OnStreamOpen(_ context.Context, id int64, typ string) error {
+	if cb.Debug {
+		log.Printf("stream %d open for %s\n", id, typ)
```

Comment:
Here and everywhere else, please start log messages with an uppercase character.